### ISSUE TITLE
Inconsistency with cursor placement at end of expansion with 2-step-converging python interpolation

### ISSUE THREAD

**Issue by hl037** [SirVer/ultisnips#1402]
There is an inconsistency on some edge cases with the cursor placement at the end of the snippet expansion 
Here two templates :

Works (cursor at the end of the expansion (the "tt") :
```
snippet test1 "test"
`!p
import sys
if 'dummy' not in sys.modules:
	sys.modules['dummy'] = 'tt'
	snip.rv=''
else:
	snip.rv='tt'
`
endsnippet
```

Problem (cursor at the first "t") :
```
snippet test2 "test"
`!p
import sys
if 'dummy' not in sys.modules:
	sys.modules['dummy'] = 'tt'
	# snip.rv is not mutated
else:
	snip.rv='tt'
`
endsnippet
```

Simple explanation : These templates expands to "tt" when the output has converged. The `if 'dummy' not in sys.modules` is just the simplest way I found (without using the new store feature I am implementing) so that the code returns an empty string on first execution, and "tt" on the second.

**Expected behavior:**

Either both the templates should put the cursor at the end, either the python code should be executed only once since rv is not changed and the snippet should be considered as having converged.

**Actual behavior:**

As stated, the first snippet puts the cursor at the end of the expantion : 
```
tt|
```

while the second put it at the the start:
```
|tt
```

**Steps to reproduce**

Add both the snippet to a snippet def file, and triger both the snippet on new lines and observe what is stated

**Additionnal Notes**

Initially, I first encountered the problem implementing #1394 But I then was able to track down the problem to `SnippetUtil._rv_changed` 

This is not a major problem since a simple workaround is to add a dummy `snip.rv=''`, but yet, it may be pertinent to document it. At least if someone come across it, this issue provides an explanation and a workaround, and I'm ok if it's tagged as "Won't fix"

**Comment by hl037** [comment]
Well... Actually, I don't really understand because I though I had found the workaround but I still have situations where despite assigning the value, I get the wrong cursor position, while al so having the test2 working as expected...

it seems it's not deterministic...

**Comment by hl037** [comment]
After having investigated #1403 I think this problem is actually the same : the Python's text object may have the same ordering values as the final Tabstop, thus, they get swapped time to times

**Comment by SirVer** [comment]
@hl037 Leo, i am super impressed with your ability of working through these difficult issues in UltiSnips! Your work is very much appreciated.
